tools/c7n_oci/c7n_oci/resources/compute.py

```diff
@@ -286,14 +281,45 @@ def process(self, resources, event):
                 comp_resources[comp_id] = [resource]
         # Query the MonitoringClient with the query against each compartment and perform
         # the filtering
+        monitoring_client = self.manager.get_session().client("oci.monitoring.MonitoringClient")
+        result = []
         for compartment_id in comp_resources.keys():
+            query = self.data.get("query")
+            filter_resources = comp_resources.get(comp_id)
+            query = InstanceMetrics.get_metrics_resource_query(query, filter_resources)
+            log.debug(
+                f"Monitoring client will execute query :{query} for resources in the compartment"
+            )
+            summarize_metrics = oci.monitoring.models.SummarizeMetricsDataDetails(
+                query=query,
+                namespace="oci_computeagent",
+            )
             metric_response = monitoring_client.summarize_metrics_data(
                 compartment_id=compartment_id,
                 summarize_metrics_data_details=summarize_metrics,
             )
-            for metric_data in metric_response.data:
+            metric_resources = metric_response.data
+            for metric_data in metric_resources:
                 resource_id = metric_data.dimensions["resourceId"]
-                for resource in comp_resources.get(compartment_id):
+                for resource in filter_resources:
                     if resource.get("id") == resource_id:
                         result.append(resource)
+
         return result
+
+    @staticmethod
+    def get_metrics_resource_query(query, filter_resources):
+        if "resourceId" not in query:
```

Comment:
this block can be demented by handling if resourceId in query: return query

Reply:
done